### ISSUE TITLE
mongodb_exporter creates a lot of noise in the MongoDB logs

### ISSUE THREAD

**Issue by arggh** [percona/mongodb_exporter#470]
Percona's MongoDB exporter causes a looooot of noise in the logs.

**To Reproduce**
1. what parameters are being passed to `mongodb_exporter`

`/usr/local/opt/mongodb_exporter-0.20.9.linux-amd64/mongodb_exporter --compatible-mode --discovering-mode --enable.top --mongodb.uri=mongodb://user:pass@localhost:27017/admin?tls=true&tlsCAFile=/etc/ssl/ca.pem&tlsCertificateKeyFile=/etc/ssl/mongo_client.pem&appName=mongodb-exporter --web.listen-address=":9216" --web.telemetry-path=/metrics --mongodb.collstats-colls=app --mongodb.indexstats-colls=app`

2. describe steps to reproduce the issue

- Deploy replica set
- Install 0.20.9 mongodb_exporter (for reasons so far unknonwn, our Prometheus server doesn't get any logs if we install 0.30.x)

**Expected behavior**
I'd expect substantially less logs being produced.

**Logs**

<details>
<summary>Sample logs during few seconds of operation</summary>

```
{"t":{"$date":"2022-04-08T08:34:51.938+00:00"},"s":"I",  "c":"ACCESS",   "id":20250,   "ctx":"conn818","msg":"Authentication succeeded","attr":{"mechanism":"SCRAM-SHA-256","speculative":true,"principalName":"monitor","authenticationDatabase":"admin","remote":"127.0.0.1:40860","extraInfo":{}}}
{"t":{"$date":"2022-04-08T08:34:52.035+00:00"},"s":"I",  "c":"-",        "id":20883,   "ctx":"conn815","msg":"Interrupted operation as its client disconnected","attr":{"opId":71365}}
{"t":{"$date":"2022-04-08T08:34:52.035+00:00"},"s":"I",  "c":"NETWORK",  "id":22944,   "ctx":"conn814","msg":"Connection ended","attr":{"remote":"127.0.0.1:40846","uuid":"4861912b-4048-40fb-a611-99245c771c19","connectionId":814,"connectionCount":31}}
{"t":{"$date":"2022-04-08T08:34:52.036+00:00"},"s":"I",  "c":"NETWORK",  "id":22944,   "ctx":"conn815","msg":"Connection ended","attr":{"remote":"127.0.0.1:40848","uuid":"d3fece6f-84e4-40d7-be99-4e7ea2349a89","connectionId":815,"connectionCount":30}}
{"t":{"$date":"2022-04-08T08:34:52.036+00:00"},"s":"I",  "c":"NETWORK",  "id":22944,   "ctx":"conn817","msg":"Connection ended","attr":{"remote":"127.0.0.1:40858","uuid":"4ec8c459-8132-480a-9200-d00bb7bf9d25","connectionId":817,"connectionCount":29}}
{"t":{"$date":"2022-04-08T08:34:52.036+00:00"},"s":"I",  "c":"NETWORK",  "id":22944,   "ctx":"conn816","msg":"Connection ended","attr":{"remote":"127.0.0.1:40852","uuid":"9ea49ea6-95a8-42e5-b4ed-91de84528601","connectionId":816,"connectionCount":28}}
{"t":{"$date":"2022-04-08T08:34:52.037+00:00"},"s":"I",  "c":"NETWORK",  "id":22944,   "ctx":"conn818","msg":"Connection ended","attr":{"remote":"127.0.0.1:40860","uuid":"2a5b65b8-6bf4-4f09-b53e-7817781e088f","connectionId":818,"connectionCount":27}}
{"t":{"$date":"2022-04-08T08:35:06.769+00:00"},"s":"I",  "c":"NETWORK",  "id":22943,   "ctx":"listener","msg":"Connection accepted","attr":{"remote":"127.0.0.1:40866","uuid":"36a2419a-c84c-4563-9669-b4036abff44c","connectionId":819,"connectionCount":28}}
{"t":{"$date":"2022-04-08T08:35:06.769+00:00"},"s":"I",  "c":"NETWORK",  "id":22943,   "ctx":"listener","msg":"Connection accepted","attr":{"remote":"127.0.0.1:40868","uuid":"af1cc45b-6a36-447f-8908-98a1fd2755c1","connectionId":820,"connectionCount":29}}
{"t":{"$date":"2022-04-08T08:35:06.798+00:00"},"s":"I",  "c":"NETWORK",  "id":51800,   "ctx":"conn820","msg":"client metadata","attr":{"remote":"127.0.0.1:40868","client":"conn820","doc":{"driver":{"name":"mongo-go-driver","version":"v1.7.2"},"os":{"type":"linux","architecture":"amd64"},"platform":"go1.16.9","application":{"name":"mongodb_exporter"}}}}
{"t":{"$date":"2022-04-08T08:35:06.802+00:00"},"s":"I",  "c":"NETWORK",  "id":22943,   "ctx":"listener","msg":"Connection accepted","attr":{"remote":"127.0.0.1:40872","uuid":"8ac2ba19-1a2d-477e-8114-1695f81bac93","connectionId":821,"connectionCount":30}}
{"t":{"$date":"2022-04-08T08:35:06.807+00:00"},"s":"I",  "c":"NETWORK",  "id":51800,   "ctx":"conn819","msg":"client metadata","attr":{"remote":"127.0.0.1:40866","client":"conn819","doc":{"driver":{"name":"mongo-go-driver","version":"v1.7.2"},"os":{"type":"linux","architecture":"amd64"},"platform":"go1.16.9","application":{"name":"mongodb_exporter"}}}}
{"t":{"$date":"2022-04-08T08:35:06.822+00:00"},"s":"I",  "c":"NETWORK",  "id":51800,   "ctx":"conn821","msg":"client metadata","attr":{"remote":"127.0.0.1:40872","client":"conn821","doc":{"driver":{"name":"mongo-go-driver","version":"v1.7.2"},"os":{"type":"linux","architecture":"amd64"},"platform":"go1.16.9","application":{"name":"mongodb_exporter"}}}}
{"t":{"$date":"2022-04-08T08:35:06.833+00:00"},"s":"I",  "c":"ACCESS",   "id":20250,   "ctx":"conn821","msg":"Authentication succeeded","attr":{"mechanism":"SCRAM-SHA-256","speculative":true,"principalName":"monitor","authenticationDatabase":"admin","remote":"127.0.0.1:40872","extraInfo":{}}}
{"t":{"$date":"2022-04-08T08:35:06.909+00:00"},"s":"I",  "c":"NETWORK",  "id":22943,   "ctx":"listener","msg":"Connection accepted","attr":{"remote":"127.0.0.1:40876","uuid":"ad82eb10-4d47-44cc-82d3-fa446567ca25","connectionId":822,"connectionCount":31}}
{"t":{"$date":"2022-04-08T08:35:06.909+00:00"},"s":"I",  "c":"NETWORK",  "id":22943,   "ctx":"listener","msg":"Connection accepted","attr":{"remote":"127.0.0.1:40882","uuid":"08ea6fa5-13b1-46b2-8699-8bdc515a67ba","connectionId":823,"connectionCount":32}}
{"t":{"$date":"2022-04-08T08:35:06.910+00:00"},"s":"I",  "c":"NETWORK",  "id":22943,   "ctx":"listener","msg":"Connection accepted","attr":{"remote":"127.0.0.1:40884","uuid":"4b5ef3ef-117f-49ac-8115-c09ec7fb5363","connectionId":824,"connectionCount":33}}
{"t":{"$date":"2022-04-08T08:35:06.952+00:00"},"s":"I",  "c":"NETWORK",  "id":51800,   "ctx":"conn823","msg":"client metadata","attr":{"remote":"127.0.0.1:40882","client":"conn823","doc":{"driver":{"name":"mongo-go-driver","version":"v1.7.2"},"os":{"type":"linux","architecture":"amd64"},"platform":"go1.16.9","application":{"name":"mongodb_exporter"}}}}
{"t":{"$date":"2022-04-08T08:35:06.959+00:00"},"s":"I",  "c":"NETWORK",  "id":51800,   "ctx":"conn824","msg":"client metadata","attr":{"remote":"127.0.0.1:40884","client":"conn824","doc":{"driver":{"name":"mongo-go-driver","version":"v1.7.2"},"os":{"type":"linux","architecture":"amd64"},"platform":"go1.16.9","application":{"name":"mongodb_exporter"}}}}
{"t":{"$date":"2022-04-08T08:35:06.960+00:00"},"s":"I",  "c":"NETWORK",  "id":51800,   "ctx":"conn822","msg":"client metadata","attr":{"remote":"127.0.0.1:40876","client":"conn822","doc":{"driver":{"name":"mongo-go-driver","version":"v1.7.2"},"os":{"type":"linux","architecture":"amd64"},"platform":"go1.16.9","application":{"name":"mongodb_exporter"}}}}
{"t":{"$date":"2022-04-08T08:35:06.960+00:00"},"s":"I",  "c":"ACCESS",   "id":20250,   "ctx":"conn822","msg":"Authentication succeeded","attr":{"mechanism":"SCRAM-SHA-256","speculative":true,"principalName":"monitor","authenticationDatabase":"admin","remote":"127.0.0.1:40876","extraInfo":{}}}
{"t":{"$date":"2022-04-08T08:35:06.960+00:00"},"s":"I",  "c":"ACCESS",   "id":20250,   "ctx":"conn823","msg":"Authentication succeeded","attr":{"mechanism":"SCRAM-SHA-256","speculative":true,"principalName":"monitor","authenticationDatabase":"admin","remote":"127.0.0.1:40882","extraInfo":{}}}
{"t":{"$date":"2022-04-08T08:35:06.961+00:00"},"s":"I",  "c":"ACCESS",   "id":20250,   "ctx":"conn824","msg":"Authentication succeeded","attr":{"mechanism":"SCRAM-SHA-256","speculative":true,"principalName":"monitor","authenticationDatabase":"admin","remote":"127.0.0.1:40884","extraInfo":{}}}
{"t":{"$date":"2022-04-08T08:35:07.047+00:00"},"s":"I",  "c":"-",        "id":20883,   "ctx":"conn820","msg":"Interrupted operation as its client disconnected","attr":{"opId":71687}}
{"t":{"$date":"2022-04-08T08:35:07.048+00:00"},"s":"I",  "c":"NETWORK",  "id":22944,   "ctx":"conn819","msg":"Connection ended","attr":{"remote":"127.0.0.1:40866","uuid":"36a2419a-c84c-4563-9669-b4036abff44c","connectionId":819,"connectionCount":32}}
{"t":{"$date":"2022-04-08T08:35:07.048+00:00"},"s":"I",  "c":"NETWORK",  "id":22944,   "ctx":"conn820","msg":"Connection ended","attr":{"remote":"127.0.0.1:40868","uuid":"af1cc45b-6a36-447f-8908-98a1fd2755c1","connectionId":820,"connectionCount":31}}
{"t":{"$date":"2022-04-08T08:35:07.048+00:00"},"s":"I",  "c":"NETWORK",  "id":22944,   "ctx":"conn821","msg":"Connection ended","attr":{"remote":"127.0.0.1:40872","uuid":"8ac2ba19-1a2d-477e-8114-1695f81bac93","connectionId":821,"connectionCount":30}}
{"t":{"$date":"2022-04-08T08:35:07.049+00:00"},"s":"I",  "c":"NETWORK",  "id":22944,   "ctx":"conn822","msg":"Connection ended","attr":{"remote":"127.0.0.1:40876","uuid":"ad82eb10-4d47-44cc-82d3-fa446567ca25","connectionId":822,"connectionCount":29}}
{"t":{"$date":"2022-04-08T08:35:07.049+00:00"},"s":"I",  "c":"NETWORK",  "id":22944,   "ctx":"conn823","msg":"Connection ended","attr":{"remote":"127.0.0.1:40882","uuid":"08ea6fa5-13b1-46b2-8699-8bdc515a67ba","connectionId":823,"connectionCount":28}}
{"t":{"$date":"2022-04-08T08:35:07.049+00:00"},"s":"I",  "c":"NETWORK",  "id":22944,   "ctx":"conn824","msg":"Connection ended","attr":{"remote":"127.0.0.1:40884","uuid":"4b5ef3ef-117f-49ac-8115-c09ec7fb5363","connectionId":824,"connectionCount":27}}
{"t":{"$date":"2022-04-08T08:35:21.767+00:00"},"s":"I",  "c":"NETWORK",  "id":22943,   "ctx":"listener","msg":"Connection accepted","attr":{"remote":"127.0.0.1:40890","uuid":"6d8985e7-06a4-4f68-ae25-2a4e1f8741ba","connectionId":825,"connectionCount":28}}
{"t":{"$date":"2022-04-08T08:35:21.767+00:00"},"s":"I",  "c":"NETWORK",  "id":22943,   "ctx":"listener","msg":"Connection accepted","attr":{"remote":"127.0.0.1:40892","uuid":"52e16e0a-25e1-4e1d-98ac-61bce4684070","connectionId":826,"connectionCount":29}}
{"t":{"$date":"2022-04-08T08:35:21.790+00:00"},"s":"I",  "c":"NETWORK",  "id":51800,   "ctx":"conn825","msg":"client metadata","attr":{"remote":"127.0.0.1:40890","client":"conn825","doc":{"driver":{"name":"mongo-go-driver","version":"v1.7.2"},"os":{"type":"linux","architecture":"amd64"},"platform":"go1.16.9","application":{"name":"mongodb_exporter"}}}}
{"t":{"$date":"2022-04-08T08:35:21.791+00:00"},"s":"I",  "c":"NETWORK",  "id":22943,   "ctx":"listener","msg":"Connection accepted","attr":{"remote":"127.0.0.1:40896","uuid":"1d057579-d5d6-470f-b93c-c6ae288311de","connectionId":827,"connectionCount":30}}
{"t":{"$date":"2022-04-08T08:35:21.806+00:00"},"s":"I",  "c":"NETWORK",  "id":51800,   "ctx":"conn826","msg":"client metadata","attr":{"remote":"127.0.0.1:40892","client":"conn826","doc":{"driver":{"name":"mongo-go-driver","version":"v1.7.2"},"os":{"type":"linux","architecture":"amd64"},"platform":"go1.16.9","application":{"name":"mongodb_exporter"}}}}
{"t":{"$date":"2022-04-08T08:35:21.813+00:00"},"s":"I",  "c":"NETWORK",  "id":51800,   "ctx":"conn827","msg":"client metadata","attr":{"remote":"127.0.0.1:40896","client":"conn827","doc":{"driver":{"name":"mongo-go-driver","version":"v1.7.2"},"os":{"type":"linux","architecture":"amd64"},"platform":"go1.16.9","application":{"name":"mongodb_exporter"}}}}
{"t":{"$date":"2022-04-08T08:35:21.824+00:00"},"s":"I",  "c":"ACCESS",   "id":20250,   "ctx":"conn827","msg":"Authentication succeeded","attr":{"mechanism":"SCRAM-SHA-256","speculative":true,"principalName":"monitor","authenticationDatabase":"admin","remote":"127.0.0.1:40896","extraInfo":{}}}
{"t":{"$date":"2022-04-08T08:35:21.912+00:00"},"s":"I",  "c":"NETWORK",  "id":22943,   "ctx":"listener","msg":"Connection accepted","attr":{"remote":"127.0.0.1:40900","uuid":"d7524e4d-0e9b-4d91-91fb-ec9e0c60a65b","connectionId":828,"connectionCount":31}}
{"t":{"$date":"2022-04-08T08:35:21.943+00:00"},"s":"I",  "c":"NETWORK",  "id":51800,   "ctx":"conn828","msg":"client metadata","attr":{"remote":"127.0.0.1:40900","client":"conn828","doc":{"driver":{"name":"mongo-go-driver","version":"v1.7.2"},"os":{"type":"linux","architecture":"amd64"},"platform":"go1.16.9","application":{"name":"mongodb_exporter"}}}}
{"t":{"$date":"2022-04-08T08:35:21.944+00:00"},"s":"I",  "c":"ACCESS",   "id":20250,   "ctx":"conn828","msg":"Authentication succeeded","attr":{"mechanism":"SCRAM-SHA-256","speculative":true,"principalName":"monitor","authenticationDatabase":"admin","remote":"127.0.0.1:40900","extraInfo":{}}}
{"t":{"$date":"2022-04-08T08:35:22.019+00:00"},"s":"I",  "c":"-",        "id":20883,   "ctx":"conn825","msg":"Interrupted operation as its client disconnected","attr":{"opId":72008}}
{"t":{"$date":"2022-04-08T08:35:22.020+00:00"},"s":"I",  "c":"NETWORK",  "id":22944,   "ctx":"conn826","msg":"Connection ended","attr":{"remote":"127.0.0.1:40892","uuid":"52e16e0a-25e1-4e1d-98ac-61bce4684070","connectionId":826,"connectionCount":30}}
{"t":{"$date":"2022-04-08T08:35:22.020+00:00"},"s":"I",  "c":"NETWORK",  "id":22944,   "ctx":"conn827","msg":"Connection ended","attr":{"remote":"127.0.0.1:40896","uuid":"1d057579-d5d6-470f-b93c-c6ae288311de","connectionId":827,"connectionCount":29}}
{"t":{"$date":"2022-04-08T08:35:22.020+00:00"},"s":"I",  "c":"NETWORK",  "id":22944,   "ctx":"conn825","msg":"Connection ended","attr":{"remote":"127.0.0.1:40890","uuid":"6d8985e7-06a4-4f68-ae25-2a4e1f8741ba","connectionId":825,"connectionCount":28}}
{"t":{"$date":"2022-04-08T08:35:22.021+00:00"},"s":"I",  "c":"NETWORK",  "id":22944,   "ctx":"conn828","msg":"Connection ended","attr":{"remote":"127.0.0.1:40900","uuid":"d7524e4d-0e9b-4d91-91fb-ec9e0c60a65b","connectionId":828,"connectionCount":27}}
```

</details>

**Environment**
 - OS: Ubuntu 20.04
 - MongoDB version 5.0.6

**Comment by ShashankSinha252** [comment]
Hi @arggh , thanks for reporting this issue.

The issue reported by you stems from the fact that **MongoDB exporter creates a new connection to MongoDB, for every metric scrape request**. As a result, you see bunch of connection establishment and teardown related logs in MongoDB. The exporter provides the option of reusing an existing connection with MongoDB by enabling global connection pool (check flag **mongodb.global-conn-pool**)

Usage of global connection pool should drop the noise in MongoDB logs. Please give it a shot and let us know if it resolves your issue.

**Comment by ShashankSinha252** [comment]
We also have a new release out for MongoDB exporter. Feel free to check it out too.

**Comment by arggh** [comment]
Thanks, I do recall trying out the connection pool flag in hopes of reducing the log bloat, but observed no significant change. I'll give it another try, maybe I messed up something (eg. forgot to reload the service).

I also did try the new release of MongoDB exporter, however, either the dashboard we are using was incompatible or due to some other issue, Grafana did not show any MongoDB data after we upgraded the latest release.

The dashboard we use is https://grafana.com/grafana/dashboards/14997

If you have any tips on how to best use Percona's MongoDB Exporter with a vanilla Grafana & Prometheus setup, I'm all ears. I did find a repository for Percona's own dashboards, but those seemed tailor-made for PMM2 and were cumbersome to install.
